modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -153,10 +153,9 @@ public URL resolve(String granuleLocation) {
                     new VectorBinarizeDescriptor(),
                     new VectorBinarizeRIF(),
                     Registry.JAI_TOOLS_PRODUCT);
-        } catch (Exception e) {
-            if (LOGGER.isLoggable(Level.FINE)) {
-                LOGGER.log(Level.FINE, e.getLocalizedMessage());
-            }
+        } catch (Throwable e) {
+            LOGGER.log(Level.SEVERE, "Error setting up GranuleDescriptor", e);
+            throw new RuntimeException("Error setting up GranuleDescriptor.", e);
         }
```

Comment:
From ignored exception to throwing? That's not backwards compatible... also, what is the relationship with the issue at hand?

Reply:
I'm willing to skip the throwing, but I would keep the logging.